src/signature.c

```diff
@@ -20,7 +21,84 @@ void signature_init(void)
 	ERR_load_crypto_strings();
 }
 
-static EVP_PKEY *load_key(const gchar *keyfile, GError **error)
+static ENGINE *get_pkcs11_engine(GError **error)
+{
+	static ENGINE *e = NULL;
+	unsigned long err;
+	const gchar *data;
+	const gchar *env;
+	int flags;
+
+	if (e)
```

Comment:
Should not happen as you just set it to NULL...

Reply:
Fixed.